lib/browser_client.dart

```diff
@@ -23,6 +23,12 @@ import 'src/response.dart';
 /// unable to directly set some headers, such as `content-length`. It is also
 /// unable to stream requests or responses; a request will only be sent and a
 /// response will only be returned once all the data is available.
+///
+/// [BrowserClient] allows setting values directly on the underlying
+/// [HttpRequest] values through [Request.context].
+///
+/// * `http.html.with_credentials` is a boolean value, defaulting to `false`
+///   that corresponds to [HttpRequest.withCredentials].
```

Comment:
I'd rather document these in terms of their semantics, rather than in terms of their implementation. Talk about what setting the context variables *do*.